framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -136,22 +150,47 @@ export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
 		return ForwardCCMsgResult.INFORM_SIDECHAIN_TERMINATION;
 	}
 
-	public async bounce(ccm: CCMsg): Promise<void> {
-		const terminatedStateAccountExists = await this.hasTerminatedStateAccount(ccm.sendingChainID);
+	public async bounce(context: CCMBounceContext): Promise<void> {
+		const { ccm, eventQueue, newCCMStatus, ccmProcessedEventCode } = context;
+		const ccmID = utils.hash(codec.encode(ccmSchema, ccm));
+		const minimumFee = MIN_RETURN_FEE * BigInt(ccmID.length);
+		if (ccm.status === CCM_STATUS_OK && ccm.fee >= minimumFee) {
+			this.events
+				.get(CcmProcessedEvent)
+				.log({ eventQueue }, ccm.sendingChainID, ccm.receivingChainID, {
+					ccmID,
+					result: CCM_PROCESSED_RESULT_BOUNCED,
+					code: ccmProcessedEventCode,
+				});
+			const newCcm = {
+				...ccm,
+				sendingChainID: ccm.receivingChainID,
+				receivingChainID: ccm.sendingChainID,
+				status: newCCMStatus,
+			};
+
+			// If the function is called during the cross-chain command execution, the fee is set to 0
+			if (newCCMStatus === CCM_STATUS_CODE_FAILED_CCM) {
+				newCcm.fee = BigInt(0);
+			} else {
+				newCcm.fee -= minimumFee;
+			}
+			await this.addToOutbox(newCcm.receivingChainID, newCcm);
```

Comment:
usually when we have abbreviation in the variable we use camelCase like `newCCM`